### PR TITLE
Tear down assocation on ABORT chunk

### PR DESCRIPTION
**Pull request by Sean-Der**
Properly tear down the association when we receive an ABORT
chunk.

### CODE REVIEW

**Comment by Sean-Der** [comment]
@enobufs I am going to write a test to confirm this doesn't regress as well, but ran out of time tonight!

File: association.go [hunk 1]
```diff
@@ -1963,8 +1965,8 @@ func (a *Association) handleChunk(p *packet, c chunk) error {
 	var err error
 
 	if _, err = c.check(); err != nil {
-		return errors.Wrap(err, "failed validating chunk")
-		// TODO: Create ABORT
```
Comment (Sean-Der) [inline association.go]: I didn't find anything in the RFC, but it seems like we should tolerate anything from the other side (besides ABORT chunks)

Comment (enobufs) [reply association.go]: There are some cases mentioned in RFC 4960 which require sending ABORT  to close the association. (grep by `should abort` or `must abort` in the RFC).
Chunks that can return an error from check() currently are chunk_init.go and chunk_init_ack.go as far as I can tell.
We've been ignoring those errors so the logic hasn't be proven. I'd support you change (just log and ignore) for now to reduce regression. We could revisit this later when we finally support "sending ABORT".

Comment (Sean-Der) [reply association.go]: Ticketed! https://github.com/pion/sctp/issues/103

**Comment by enobufs** [comment]
@Sean-Der Just to be sure that it can handle multiple errorCauses, I have added a marshaling test. I think your changes look good!

**Comment by Sean-Der** [comment]
@enobufs thank you so much. I appreciate how much work you have put forward :)

**Pull request merged**